pkg/kubelet/cm/container_manager_linux_test.go

```diff
@@ -268,3 +268,65 @@ func TestGetCapacity(t *testing.T) {
 		})
 	}
 }
+
+func TestNewPodContainerManager(t *testing.T) {
+	cases := []struct {
+		name string
+		cm   *containerManagerImpl
+	}{
+		{
+			name: "CgroupsPerQOS is disabled, return *podContainerManagerNoop",
+			cm: &containerManagerImpl{
+				qosContainerManager: &qosContainerManagerImpl{
+					qosContainersInfo: QOSContainersInfo{
+						Guaranteed: CgroupName{"guaranteed"},
+						BestEffort: CgroupName{"besteffort"},
+						Burstable:  CgroupName{"burstable"},
+					},
+					subsystems:         &CgroupSubsystems{},
+					cgroupManager:      NewCgroupManager(&CgroupSubsystems{}, ""),
+					activePods:         func() []*v1.Pod { return []*v1.Pod{} },
+					getNodeAllocatable: func() v1.ResourceList { return v1.ResourceList{} },
+					cgroupRoot:         CgroupName{"burstable"},
+					qosReserved:        map[v1.ResourceName]int64{},
+				},
+
+				NodeConfig: NodeConfig{
+					CgroupsPerQOS: false,
+				},
+			},
+		},
+		{
+			name: "CgroupsPerQOS is enabled, return *podContainerManagerImpl",
+			cm: &containerManagerImpl{
+				qosContainerManager: &qosContainerManagerImpl{
```

Comment:
qosContainerManager seems to be the same for both use cases. Can you reuse it?

Reply:
thanks for the pointer. Added as a variable for reuse as well as QOS node configs